sql/parser/builtins.go

```diff
@@ -1161,6 +1186,31 @@ func regexpEvalFlags(pattern, sqlFlags string) (string, error) {
 	return fmt.Sprintf("(?%s:%s)", flagString.String(), pattern), nil
 }
 
+func overlay(s, to string, pos, size int) (Datum, error) {
+	if pos < 1 {
+		return nil, errors.New("negative substring length not allowed")
+	}
+
+	var out bytes.Buffer
+	i := 0
+	for _, c := range s {
+		if i >= pos-1 {
+			break
+		}
+		out.WriteRune(c)
+		i++
+	}
```

Comment:
I'm not quite seeing how that would simplify things because the loops have a slightly different structure. One writes up to a given character count, while the other starts writing past a given character count.